src/sentry/sentry_metrics/indexer/cache.py

```diff
@@ -0,0 +1,78 @@
+import logging
+import random
+from typing import Mapping, MutableMapping, Optional, Sequence
+
+from django.conf import settings
+
+from sentry.utils.cache import cache
+from sentry.utils.hashlib import md5_text
+
+logger = logging.getLogger(__name__)
+
+
+class StringIndexerCache:
```

Comment:
Please have a unit test for this class in isolation. 
It will provide three advantages over relying on the indexer test to test the cache:
1) You can actually test the cache with confidence. If somebody breaks the indexer in such a way that it does not use the cache anymore, the indexer test would not be testing the cache anymore. And you have to rely on the indexer test to look in the internals of the indexer to make assertions regarding the state of the cache. This is kind of inevitable but a bad practice. If we keep our tests black box, we make them more resilient to refactoring.
2) You can test corner cases 
3) You make it easier to understand what is broken when you see test failure. If the cache test fails then the cache is broken. if you have only the indexer test, if that test fails, either the indexer or the cache are broken and you have to investigate for longer.